### PR TITLE
MAINT: Add a check to verify all `.pyi` files are installed by meson

### PR DESCRIPTION
**Pull request by Smit-create**
Reference: https://github.com/scipy/scipy/pull/15476#issuecomment-1102479430

cc @rgommers

### CODE REVIEW

**Comment by Smit-create** [comment]
The test should fail since the following files are missing:
```
/home/admin-pc/Smitlunagariya/scipy/scipy/stats/_qmc_cy.pyi
/home/admin-pc/Smitlunagariya/scipy/scipy/stats/_sobol.pyi
/home/admin-pc/Smitlunagariya/scipy/scipy/linalg/_matfuncs_expm.pyi
/home/admin-pc/Smitlunagariya/scipy/scipy/linalg/_cythonized_array_utils.pyi
/home/admin-pc/Smitlunagariya/scipy/scipy/optimize/__nnls.pyi
/home/admin-pc/Smitlunagariya/scipy/scipy/special/_ufuncs.pyi
/home/admin-pc/Smitlunagariya/scipy/scipy/special/_test_round.pyi
/home/admin-pc/Smitlunagariya/scipy/scipy/special/cython_special.pyi
/home/admin-pc/Smitlunagariya/scipy/scipy/special/_orthogonal.pyi
/home/admin-pc/Smitlunagariya/scipy/scipy/spatial/_ckdtree.pyi
/home/admin-pc/Smitlunagariya/scipy/scipy/spatial/_voronoi.pyi
/home/admin-pc/Smitlunagariya/scipy/scipy/spatial/distance.pyi
/home/admin-pc/Smitlunagariya/scipy/scipy/spatial/_qhull.pyi
```

**Comment by rgommers** [comment]
Thanks @Smit-create. Yes, the test indeed fails with the expected message.

Will you install the files in the next commit? Note that for generated files, I figured out a solution once that you can adapt (from `scipy/special/meson.build`):
```python
# Must use `custom_target`, because `py3.install_sources` does not work with
# generated sources - see https://github.com/mesonbuild/meson/issues/7372
...
endforeach
```

**Comment by Smit-create** [comment]
> Will you install the files in the next commit?

Yeah, will fix that.

**Comment by Smit-create** [comment]
> The test should fail since the following files are missing:

The list had a couple of extra files because of path suffix mismatch which is now fixed.
Example:
Scipy repo: `scipy/scipy/spatial/_voronoi.pyi`
Meson installed: `site-packages/scipy/spatial/_voronoi.pyi`

So, I fixed that by picking only the last 2 parent directories of the original file instead of 3.

File: scipy/special/meson.build [hunk 1]
```diff
@@ -519,12 +524,5 @@ py3.install_sources(
   subdir: 'scipy/special'
 )
 
-# FIXME: fails with "Argument must be string or file."
-#py3.install_sources(
```
Comment (Smit-create) [inline scipy/special/meson.build:523]: This looks good to review now. The FIXME note also seems to be resolved in this diff.

Comment (Smit-create) [inline scipy/special/meson.build:523]: This looks good to review now. The FIXME note also seems to be resolved in this diff.

**Review by rgommers — APPROVED** [review]
LGTM now, thanks @Smit-create. We're now installing some extra (unwanted) files, so I added some comments on options to deal with that in the future. It's not a real issue, so the best thing is to fix things in Meson so `py3.install_sources` can accept generated sources.

**Comment by Smit-create** [comment]
Thank you!!

**Pull request merged**